drake/solvers/OptimalMotionPlanningTree.m

```diff
@@ -0,0 +1,342 @@
+classdef OptimalMotionPlanningTree < TaskSpaceMotionPlanningTree
```

Comment:
It feels like perhaps this class should be merged with MotionPlanningTree so as to give us RRT* for all the children. Obviously that would take a bit of work, to let the rrt and rrtStar capabilities live side by side. If we aren't going to do that now, this class should be renamed to `OptimalTaskSpaceMotionPlanningTree` so as not to be misleading.